### PR TITLE
Damage zone layout refactor

### PR DESCRIPTION
**Pull request by jmgoldb**
damage zones look really pretty now!

### CODE REVIEW

**Comment by jmgoldb** [comment]
Can i get a quick code review, please? @vokal/ios-developers

File: MH4U Dex/monsterDamage.json [hunk 1]
```diff
@@ -2071,7 +2071,7 @@
   },
   {
     "_id":139,
-    "part":"Ass",
```
Comment (seanwolter) [inline MH4U Dex/monsterDamage.json:2074]: this change caught me off guard and cracked me up

File: MH4U Dex/DamageZonesCollectionViewController.m [hunk 2]
```diff
@@ -36,45 +31,50 @@ @implementation DamageZonesCollectionViewController
...
 
 - (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section
 {
-    switch (section) {
-        case DamageZoneLegendSection:
-            return 1;
-        default:
-            return self.damageZones.count;
-    }
+    return (self.damageZones.count + 1) * NumberOfCategories;
```
Comment (seanwolter) [inline MH4U Dex/DamageZonesCollectionViewController.m:58]: Sorry, why the + 1?

File: MH4U Dex/DamageZonesCollectionViewController.m [hunk 3]
```diff
@@ -36,45 +31,50 @@ @implementation DamageZonesCollectionViewController
...
-        damageZone = (DamageZone *)self.damageZones[indexPath.row];
-    }
-    switch (indexPath.section) {
-        case DamageZoneLegendSection:
-            // Don't do anything for the first cell.
-            break;
-        default:
-            // Change the cell's label
-            [cell displayContentsWithDamageZone:damageZone];
+        damageZone = (DamageZone *)self.damageZones[indexPath.row/NumberOfCategories - 1];
```
Comment (seanwolter) [inline MH4U Dex/DamageZonesCollectionViewController.m:73]: Is the cast to `DamageZone` required? I thought literal syntax returned an `id`

File: MH4U Dex/DamageZonesCollectionViewController.m [hunk 4]
```diff
@@ -36,45 +31,50 @@ @implementation DamageZonesCollectionViewController
...
-    self.flowLayout.itemSize = CGSizeMake(CGRectGetWidth(self.collectionView.bounds), 40.0f);
+
     self.damageZones = [self.monster.damageZone sortedArrayUsingDescriptors:@[[NSSortDescriptor
                                                                                sortDescriptorWithKey:DamageZoneAttributes.id
                                                                                ascending:YES]]];
 }
 
+- (void)viewWillLayoutSubviews
+{
+    CGFloat cellSize = CGRectGetWidth(self.collectionView.bounds);
```
Comment (seanwolter) [inline MH4U Dex/DamageZonesCollectionViewController.m:42]: Should/must you call `super viewWillLayoutSubviews` here?
What does this cellSize wiggling accomplish?

File: MH4U Dex/monsterDamage.json (hunk 1 above)
Comment (jmgoldb) [inline MH4U Dex/monsterDamage.json:2074]: So Rude, so Unprofessional....

File: MH4U Dex/DamageZonesCollectionViewController.m (hunk 2 above)
Comment (jmgoldb) [inline MH4U Dex/DamageZonesCollectionViewController.m:58]: I want an extra 10 cells to act as the Legend/Header

File: MH4U Dex/DamageZonesCollectionViewController.m (hunk 3 above)
Comment (jmgoldb) [inline MH4U Dex/DamageZonesCollectionViewController.m:73]: You're right, it's not!

File: MH4U Dex/DamageZonesCollectionViewController.m (hunk 4 above)
Comment (jmgoldb) [inline MH4U Dex/DamageZonesCollectionViewController.m:42]: The documentation said viewWillLayoutSubviews doesn't do anything by default, so I didn't really think it was necessary.  I could add it, though.

The cellSize wiggling makes it so there are exactly 5 cells on every row, and that their sizes change dynamically based on the screen width.  The position and layout of the cells is deliberate, so there always need to be 5 cells per row.

I could probably have turned this into a tableView instead, but to be honest, I wanted to see how I would go about doing it in a collectionView.

Comment (seanwolter) [inline MH4U Dex/DamageZonesCollectionViewController.m:42]: I'd recommend adding a comment to almost all layout code to remind Future-You what you Present-You was thinking.

**Comment by seanwolter** [comment]
Good to pull

**Pull request merged**